src/LSST/control/NetCommandDefs.cpp

```diff
@@ -73,6 +76,91 @@ bool NCmdSwitchCommandSource::action() {
     return result;
 }
 
+
+NCmdPower::Ptr NCmdPower::create(JsonPtr const& inJson_) {
+    auto cmd = Ptr(new NCmdPower(inJson_));
+    return cmd;
+}
+
+NCmdPower::NCmdPower(JsonPtr const& inJson) : NetCommand(inJson) {
+    LTRACE("NCmdPower::NCmdPower ", to_string(*inJson));
+    try {
+        int powerType = inJson->at("powerType");
```

Comment:
Such name `powerType` may be unsafe as it's very close to the data member `_powerType`. 

Besides, I would suggest putting the whole translation into a separate function in the anonymous namespace:
```
namespace {
NCmdPower::PowerSystemType translatePowerType(shared_ptr<json> const& inJson) {
    int const powerType = inJson->at("powerType");
    switch (powerType) {
        case NCmdPower::PowerSystemType::MOTOR: return NCmdPower::PowerSystemType::MOTOR;
        case NCmdPower::PowerSystemType::COMM: return NCmdPower::PowerSystemType:::COMM;
        default:
            throw NetCommandException(ERR_LOC, "unknown powerType=" + to_string(powerType) + " in " + inJson->dump());
    }
}
}
```
After that, you will be able to do this:
```
NCmdPower::NCmdPower(JsonPtr const& inJson) : NetCommand(inJson) {
    LTRACE("NCmdPower::NCmdPower ", to_string(*inJson));
    try {
        _powerType = ::translatePowerType(inJson);
        _status = inJson->at("status");
       ...
```
And your code becomes less clogged and observable.

Reply:
Changing the name.  
I'd say there's a pretty good argument for putting the conversion from int -> enum PowerSystemType with the definition of PowerSystemType. I suspect that will be needed again.